### PR TITLE
Add commit message convention

### PR DESCRIPTION
**Pull request by acm19**
Adds a detailed explanation of how commits should be written. Adds a
recipe to the `Makefile` to run the tests.

Resolves: https://github.com/acm19/aws-request-signing-apache-interceptor/issues/47

### Pull Request Checklist:

- [x] I have added a contribution line at the top of [CHANGELOG.md](CHANGELOG.md).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,9 @@
-RELEASE_BRANCH=release
```
Comment (dblock) [inline Makefile:1]: This seems unrelated, remove?

**Review by dblock — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (acm19) [reply Makefile]: Moved to a separate PR: https://github.com/acm19/aws-request-signing-apache-interceptor/pull/72
